osu.Game/Screens/Ranking/Expanded/StarRatingDisplay.cs

```diff
@@ -22,86 +22,117 @@ namespace osu.Game.Screens.Ranking.Expanded
     /// </summary>
     public class StarRatingDisplay : CompositeDrawable
     {
-        private readonly StarDifficulty difficulty;
+        [Resolved]
```

Comment:
What's the reasoning for any change in this file? Please revert all changes unless there's valid reason (I cannot see any).

Reply:
The reasoning behind it is discussed in this conversation: https://github.com/ppy/osu/pull/12503#discussion_r618753148
The `StarRatingDisplay` was recreated as a whole before because the whole wedge was redrawn, it still needs to be updated and have a transition and therefore it was discussed how we want to accomplish that

Reply:
It still is recreated every beatmap change in your code.

Reply:
Yes but it also needs to change when the user changes the mod (Double-Time f.e.) to adjust to the new difficulty it creates. Before the whole wedge got recreated on a mod changes this does not happen anymore

Reply:
Want to split that out into its own change? Judging by the issues I found with it, there's too much being changed in this PR.